### PR TITLE
Data 2449/upload and train

### PR DESCRIPTION
**Pull request by etai-shuchatowitz**
JIRA: https://viam.atlassian.net/browse/DATA-2449?atlOrigin=eyJpIjoiMWY2NDFhNzMxMmY0NDgzZmIyZjQ2OTJiNzBiNWY5YjAiLCJwIjoiaiJ9

This PR creates one cli command to upload a custom script and submit a job with that new script. 

Example: ```viam train submit-custom --dataset-id=66391a0699a6c7aa4a33e43e --org-id=3977fd7d-b1c6-4f46-9842-0c870f374d3c --model-name=model-test --path=test.txt --script-name=etai-dummy-test```

## Testing
Tested this with ```train submit-custom --dataset-id=66391a0699a6c7aa4a33e43e --org-id=3977fd7d-b1c6-4f46-9842-0c870f374d3c --model-name=model-test --path=test.txt --script-name=etai-dummy-test```

and got the logs

> succesfully uploaded training script to 3977fd7d-b1c6-4f46-9842-0c870f374d3c:etai-dummy-test
> Submitted training job with ID 663921c9e7a6d10ff1746884

Further verified in MongoDB that a record exists

```
{
  "_id": {
    "$oid": "663921c9e7a6d10ff1746885"
  },
  "metadata": {
    "request": null,
    "_id": "663921c9e7a6d10ff1746884",
    "dataset_id": "66391a0699a6c7aa4a33e43e",
    "organization_id": "3977fd7d-b1c6-4f46-9842-0c870f374d3c",
    "model_name": "model-test",
    "model_version": "2024-05-06T14-30-33",
    "model_type": 0,
    "model_framework": 0,
    "is_custom_job": true,
    "registry_item_id": "3977fd7d-b1c6-4f46-9842-0c870f374d3c:etai-dummy-test",
    "status": 5,
    "error_status": null,
    "created_on": {
      "$date": "2024-05-06T18:30:33.362Z"
    },
    "last_modified": {
      "$date": "2024-05-06T18:30:41.351Z"
    },
    "training_started": null,
    "training_ended": null,
    "synced_model_id": "",
    "tags": null
  },
  "output_path": "3977fd7d-b1c6-4f46-9842-0c870f374d3c/model-test/2024-05-06T14-30-33/2024-05-06T18:30:33.362820181Z",
  "cancelation_requested": null,
  "vertex_job_canceled": {
    "$date": "2024-05-06T18:30:41.351Z"
  },
  "vertex_job_id": "",
  "processing": false,
  "user_info": null
}
```

### CODE REVIEW

File: cli/ml_training.go [hunk 1]
```diff
@@ -9,15 +9,32 @@ import (
 	"google.golang.org/protobuf/types/known/structpb"
 )
 
-// MLTrainingUploadAction retrieves the logs for a specific build step.
+// MLTrainingUploadAction uploads a new custom training script
```
Comment (tahiyasalam) [inline cli/ml_training.go:12]: oops, thank you 😅

File: cli/train.go [hunk 2]
```diff
@@ -23,6 +24,31 @@ const (
 	trainingStatusPrefix = "TRAINING_STATUS_"
 )
 
+// DataSubmitCustomTrainingJob is the corresponding action for 'data train submit-custom'.
+func DataSubmitCustomTrainingJob(c *cli.Context) error {
```
Comment (tahiyasalam) [inline cli/train.go:28]: Sorry if this was originally implemented by me, but I think this and below should be MLSubmit and the action is just `viam train`

**Review by tahiyasalam — CHANGES_REQUESTED** [review]
Just a couple questions and nit comments!

File: cli/train.go (hunk 2 above)
Comment (etai-shuchatowitz) [reply cli/train.go:28]: Oh, right! You are 100% correct 🤦

File: cli/app.go [hunk 3]
```diff
@@ -858,6 +858,60 @@ var app = &cli.App{
...
+							Required: true,
+							Usage:    "organization ID to upload and run training job",
+						},
+						&cli.StringFlag{
+							Name:     mlTrainingFlagName,
+							Usage:    "script name of the ML training script to upload",
+							Required: true,
+						},
+						&cli.StringFlag{
+							Name:     mlTrainingFlagVersion,
```
[the remark this answers is not in the document]
Comment (etai-shuchatowitz) [reply cli/app.go:898]: I believe this happens behind the scenes https://github.com/viamrobotics/rdk/blob/main/cli/packages.go/#L174-L175 (this is just the exact same functionality as the other one)

[the remark this answers is not in the document]
Comment (etai-shuchatowitz) [reply cli/app.go:900]: Added a note to the Usage to clarify

File: cli/train.go [hunk 4]
```diff
@@ -68,6 +94,31 @@ func (c *viamClient) dataSubmitTrainingJob(datasetID, orgID, modelName, modelVer
 	return resp.Id, nil
 }
 
+// dataSubmitTrainingJob trains on data with the specified filter.
+func (c *viamClient) dataSubmitCustomTrainingJob(datasetID, registryItemID, orgID, modelName,
```
Comment (tahiyasalam) [inline cli/train.go:98]: I think we can change this to mlSubmitCustomTrainingJob and then the description would be to train on data with a specified dataset and registry item

**Review by tahiyasalam — APPROVED** [review]
One more small thing! Thanks for doing this so quickly 🚀

File: cli/train.go (hunk 4 above)
Comment (etai-shuchatowitz) [reply cli/train.go:98]: Gah! 🥚 🫤 

For those at home trying to follow along -- this means egg on face.

**Pull request merged**